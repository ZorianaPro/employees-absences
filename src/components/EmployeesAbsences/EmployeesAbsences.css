@import url('https://fonts.googleapis.com/css2?family=Dancing+Script:wght@600&family=Roboto+Slab:wght@300;400&display=swap');
:root {
  --colorBlack: #000;
  --colorWhite: #fff;
  --colorRaisinBlack: #2A2D34;

  --fontDancing: 'Dancing Script', cursive;
  --fontRoboto: 'Roboto Slab', serif;

  --fontSize4: .125em;
  --fontSize8: .25em;
  --fontSize12: .75em;
  --fontSize14: .875em;
  --fontSize16: 1em;
  --fontSize18: 1.125em;
  --fontSize20: 1.25em;
  --fontSize24: 1.5em;
  --fontSize26: 1.62em;
  --fontSize30: 1.875em;
  --fontSize36: 2.25em;
  --fontSize48: 4em;
}

.EmployeesAbsences {
  position: relative;
  width: 100%;
  height: 100%;
  min-height: 100vh;
  max-width: 998px;
  background: var(--colorWhite);
  color: var(--colorRaisinBlack);
  font-family: var(--fontRoboto);
  margin: 0 auto;
  justify-content: flex-start;
  align-content: flex-start;
  display: flex;
  flex-direction: column;
}

.EmployeesAbsences .EmployeesAbsences-Container {
  position: relative;
  height: 100%;
  width: 100%;
  padding: 2.5em 0 9em;
  display: flex;
}

.EmployeesAbsences .react-calendar {
  width: 30em;
  max-height: 20em;
}