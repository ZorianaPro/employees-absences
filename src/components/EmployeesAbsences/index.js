import React, { useEffect, useState, useCallback } from 'react';
import absencesService from '../../services/absences';
import membersService from '../../services/members';
import Timeline  from 'react-calendar-timeline';
import iCalendarService from '../../services/iCalendar';
import moment from 'moment';
import 'react-calendar-timeline/lib/Timeline.css';
import './EmployeesAbsences.css';

const EmployeesAbsences = () => {
  const [absences, setAbsences] = useState([]);
  const [members, setMembers] = useState([]);

  useEffect(() => {
    try {
      const responceAbsences = absencesService.listWithMoment();
      const responceMembers = membersService.list();
      setAbsences(responceAbsences);
      setMembers(responceMembers);
    } catch (error) {
      console.log(error);
    }
  }, []);

  const onClick = useCallback(async (event) => {
    try {
      const absencesToExport = await absencesService.listWithMembers(members);
      console.log(absencesToExport);
      iCalendarService.link({
        absences: absencesToExport,
        filename: 'download.ics'
      });
    } catch (error) {
      console.log(error);
    }
  }, [members]);

  return (
    <div className="EmployeesAbsences">
      <div onClick={ onClick }>
        Add to Calendar
      </div>
      <div className="Members-Table">
        <div>
          <Timeline
            groups={ members }
            items={ absences }
            defaultTimeStart={ moment([2017, 0, 1]) }
            defaultTimeEnd={ moment([2017, 0, 1]).add(1, 'year') }
            keys={{
              groupIdKey: 'userId',
              groupTitleKey: 'name',
              groupRightTitleKey: 'name',
              groupLabelKey: 'name',
              itemIdKey: 'id',
              itemTitleKey: 'type',
              itemDivTitleKey: 'type',
              itemGroupKey: 'userId',
              itemTimeStartKey: 'startDate',
              itemTimeEndKey: 'endDate'
            }}
            canMove={ false }
            canResize={ false }
          />
        </div>
      </div>
    </div>
  );
};

export default EmployeesAbsences;