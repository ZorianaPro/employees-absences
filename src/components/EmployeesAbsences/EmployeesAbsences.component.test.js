import React from 'react';
import { shallow } from 'enzyme';
import { MemoryRouter as Router } from 'react-router-dom';
import EmployeesAbsences from './';

describe('EmployeesAbsences', () => {
  let employeesAbsences;
  beforeEach(() => {
    employeesAbsences = shallow(
      <Router>
        <EmployeesAbsences/>
      </Router>
    );
  });
  it('renders without crashing', () => {
    expect(shallow.bind(shallow,
      <Router>
        <EmployeesAbsences/>
      </Router>
    )).not.toThrow();
  });

  it('renders as expected', () => {
    expect(employeesAbsences.find('.EmployeesAbsences'))
      .toMatchSnapshot();
  });
});
