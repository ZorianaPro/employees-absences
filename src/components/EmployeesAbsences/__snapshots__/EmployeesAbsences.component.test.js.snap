// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmployeesAbsences renders as expected 1`] = `
<div
  className="EmployeesAbsences"
>
  <div
    onClick={[Function]}
  >
    Add to Calendar
  </div>
  <div
    className="Members-Table"
  >
    <div>
      <ReactCalendarTimeline
        canChangeGroup={true}
        canMove={false}
        canResize={false}
        canSelect={true}
        className=""
        clickTolerance={3}
        dayBackground={null}
        defaultTimeEnd={"2017-12-31T23:00:00.000Z"}
        defaultTimeStart={"2016-12-31T23:00:00.000Z"}
        dragSnap={900000}
        groups={Array []}
        headerLabelFormats={
          Object {
            "dayLong": "dddd, LL",
            "dayShort": "L",
            "hourLong": "dddd, LL, HH:00",
            "hourMedium": "HH:00",
            "hourMediumLong": "L, HH:00",
            "hourShort": "HH",
            "monthLong": "MMMM YYYY",
            "monthMedium": "MM/YYYY",
            "monthMediumLong": "MMM YYYY",
            "monthShort": "MM/YY",
            "time": "LLL",
            "yearLong": "YYYY",
            "yearShort": "YY",
          }
        }
        headerRef={[Function]}
        horizontalLineClassNamesForGroup={null}
        itemHeightRatio={0.65}
        itemTouchSendsClick={false}
        items={Array []}
        keys={
          Object {
            "groupIdKey": "userId",
            "groupLabelKey": "name",
            "groupRightTitleKey": "name",
            "groupTitleKey": "name",
            "itemDivTitleKey": "type",
            "itemGroupKey": "userId",
            "itemIdKey": "id",
            "itemTimeEndKey": "endDate",
            "itemTimeStartKey": "startDate",
            "itemTitleKey": "type",
          }
        }
        lineHeight={30}
        maxZoom={157783680000}
        minResizeWidth={20}
        minZoom={3600000}
        moveResizeValidator={null}
        onBoundsChange={null}
        onCanvasClick={null}
        onItemClick={null}
        onItemContextMenu={null}
        onItemDeselect={null}
        onItemDoubleClick={null}
        onItemDrag={null}
        onItemMove={null}
        onItemResize={null}
        onItemSelect={null}
        onTimeChange={[Function]}
        onZoom={null}
        rightSidebarWidth={0}
        scrollRef={[Function]}
        selected={null}
        sidebarWidth={150}
        stackItems={false}
        stickyHeader={true}
        style={Object {}}
        subHeaderLabelFormats={
          Object {
            "dayLong": "dddd, Do",
            "dayMedium": "dd D",
            "dayMediumLong": "ddd, Do",
            "dayShort": "D",
            "hourLong": "HH:00",
            "hourShort": "HH",
            "minuteLong": "HH:mm",
            "minuteShort": "mm",
            "monthLong": "MMMM",
            "monthMedium": "MMM",
            "monthShort": "MM",
            "yearLong": "YYYY",
            "yearShort": "YY",
          }
        }
        timeSteps={
          Object {
            "day": 1,
            "hour": 1,
            "minute": 1,
            "month": 1,
            "second": 1,
            "year": 1,
          }
        }
        traditionalZoom={false}
        useResizeHandle={false}
        verticalLineClassNamesForTime={null}
        visibleTimeEnd={null}
        visibleTimeStart={null}
      />
    </div>
  </div>
</div>
`;
